core/src/executor/select/mod.rs

```diff
@@ -166,93 +171,141 @@ pub async fn select_with_labels<'a, T>(
-            filter
-                .check(Rc::clone(&blend_context))
-                .await
-                .map(|pass| pass.then(|| blend_context))
-        }
-    });
+                        Ok((table_alias, columns))
+                    }
+                })
+                .try_collect::<Vec<_>>()
+                .await?;
+
+            let labels = if with_labels {
+                get_labels(projection, table.get_alias(), &columns, &join_columns)?
+            } else {
+                vec![]
+            };
+
+            let columns = Rc::from(columns);
+            let join_columns = join_columns
+                .into_iter()
+                .map(|(_, columns)| columns)
+                .map(Rc::from)
+                .collect::<Vec<_>>();
+
+            let join = Join::new(
+                storage,
+                joins,
+                join_columns,
+                filter_context.as_ref().map(Rc::clone),
+            );
+
+            let aggregate = Aggregator::new(
+                storage,
+                projection,
+                group_by,
+                having.as_ref(),
+                filter_context.as_ref().map(Rc::clone),
+            );
+            let blend = Rc::new(Blend::new(
+                storage,
+                filter_context.as_ref().map(Rc::clone),
+                projection,
+            ));
+            let filter = Rc::new(Filter::new(
+                storage,
+                where_clause.as_ref(),
+                filter_context.as_ref().map(Rc::clone),
+                None,
+            ));
+            let limit = Limit::new(query.limit.as_ref(), query.offset.as_ref())?;
+            let sort = Sort::new(storage, filter_context, order_by);
+
+            let rows = fetch_blended(storage, table, columns).await?;
+            let rows = join.apply(rows).await?;
+            let rows = rows.try_filter_map(move |blend_context| {
+                let filter = Rc::clone(&filter);
+
+                async move {
+                    filter
+                        .check(Rc::clone(&blend_context))
+                        .await
+                        .map(|pass| pass.then(|| blend_context))
+                }
+            });
 
-    let rows = aggregate.apply(rows).await?;
-    let rows = sort
-        .apply(rows)
-        .await?
-        .and_then(move |(aggregated, context)| {
-            let blend = Rc::clone(&blend);
+            let rows = aggregate.apply(rows).await?;
+            let rows = sort
+                .apply(rows)
+                .await?
+                .and_then(move |(aggregated, context)| {
+                    let blend = Rc::clone(&blend);
 
-            async move { blend.apply(aggregated, context).await }
-        });
-    let rows = limit.apply(rows);
+                    async move { blend.apply(aggregated, context).await }
+                });
+            let rows = limit.apply(rows);
 
-    Ok((labels, rows))
+            Ok((labels, rows))
```

Comment:
cannot return value referencing temporary value
returns a value referencing data owned by the current functionrustc[E0515](https://doc.rust-lang.org/error-index.html#E0515)
mod.rs(267, 28): temporary value created here